addons/project/models/project.py

```diff
@@ -1103,6 +1104,68 @@ def get_empty_list_help(self, help):
         )
         return super(Task, self).get_empty_list_help(help)
 
+    def _valid_field_parameter(self, field, name):
+        # If the field has `task_dependency_tracking` on we track the changes made in the dependent task on the parent task
+        return name == 'task_dependency_tracking' or super()._valid_field_parameter(field, name)
+
+    @tools.ormcache('self.env.uid', 'self.env.su')
+    def _get_depends_tracked_fields(self):
+        """ Returns the set of tracked field names for the current model.
+        Those fields are the ones tracked in the parent task when using task dependencies.
+
+        See :meth:`mail.models.MailThread._get_tracked_fields`"""
+        fields = {name for name, field in self._fields.items() if getattr(field, 'task_dependency_tracking', None)}
+        return fields and set(self.fields_get(fields))
+
+    @api.model
+    def _dependency_finalize_tracking(self):
+        '''
+        Generate the tracking messages for the records that have been prepared in :meth:`write`
+        See :meth:`mail.models.MailThread._finalize_tracking`
+        '''
+        tracking_values = self.env.cr.precommit.data.pop('project.task.dependency.tracking', {})
+        parent_ids = list(tracking_values.get('dependencies', {}).keys())
+        child_ids = list(tracking_values.keys() ^ {'dependencies'})
+        if not parent_ids:
+            return
+        context = clean_context(self._context)
+        parent_tasks = self.env['project.task'].browse(parent_ids).with_context(context)
```

Comment:
```suggestion
        parent_tasks = self.env['project.task'].with_context(context).browse(parent_ids)
```
To avoid rebrowsing the records.